reflex/experimental/vars/base.py

```diff
@@ -452,3 +465,241 @@ def create(
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class LiteralBooleanVar(LiteralVar):
+    """Base class for immutable literal boolean vars."""
+
+    _var_value: bool = dataclasses.field(default=False)
+
+    @classmethod
+    def create(
+        cls,
+        value: bool,
+        _var_data: VarData | None = None,
+    ) -> LiteralBooleanVar:
+        """Create a var from a boolean value.
+
+        Args:
+            value: The value to create the var from.
+            _var_data: Additional hooks and imports associated with the Var.
+
+        Returns:
+            The var.
+        """
+        return LiteralBooleanVar(
+            _var_name="true" if value else "false",
+            _var_data=ImmutableVarData.merge(_var_data),
+            _var_type=bool,
+            _var_value=value,
+        )
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class LiteralNumberVar(LiteralVar):
+    """Base class for immutable literal number vars."""
+
+    _var_value: float | int = dataclasses.field(default=0)
+
+    @classmethod
+    def create(
+        cls,
+        value: float | int,
+        _var_data: VarData | None = None,
+    ) -> LiteralNumberVar:
+        """Create a var from a number value.
+
+        Args:
+            value: The value to create the var from.
+            _var_data: Additional hooks and imports associated with the Var.
+
+        Returns:
+            The var.
+        """
+        return LiteralNumberVar(
+            _var_name=str(value),
+            _var_data=ImmutableVarData.merge(_var_data),
+            _var_type=float | int,
+            _var_value=value,
+        )
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class LiteralObjectVar(LiteralVar):
+    """Base class for immutable literal object vars."""
+
+    _var_value: Tuple[Tuple[str, Var], ...] = dataclasses.field(default_factory=tuple)
+
+    def __init__(
+        self,
+        _var_value: dict[str, Var],
+        _var_data: VarData | None = None,
+    ):
+        """Initialize the object var.
+
+        Args:
+            _var_value: The value of the var.
+            _var_data: Additional hooks and imports associated with the Var.
+        """
+        super(LiteralObjectVar, self).__init__(
+            _var_name="",
+            _var_type=dict,
+            _var_data=ImmutableVarData.merge(_var_data),
+        )
+        object.__setattr__(self, "_var_value", tuple(_var_value.items()))
+        object.__delattr__(self, "_var_name")
+
+    def __getattr__(self, name):
+        """Get an attribute of the var.
+
+        Args:
+            name: The name of the attribute.
+
+        Returns:
+            The attribute of the var.
+        """
+        if name == "_var_name":
+            return self._cached_var_name
+        return getattr(super(), name)
```

Comment:
doesn't seem like this actually works. i had to use `return super(type(self), self).__getattr__(name)`

> slots: If true (the default is False), [__slots__](https://docs.python.org/3/reference/datamodel.html#object.__slots__) attribute will be generated and new class will be returned instead of the original one. If __slots__ is already defined in the class, then [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) is raised. Calling no-arg [super()](https://docs.python.org/3/library/functions.html#super) in dataclasses using slots=True will result in the following exception being raised: TypeError: super(type, obj): obj must be an instance or subtype of type. The two-arg [super()](https://docs.python.org/3/library/functions.html#super) is a valid workaround. See [gh-90562](https://github.com/python/cpython/issues/90562) for full details. [[1]](https://docs.python.org/3/library/dataclasses.html)